File: angularapp/src/app/_models/recipe.ts
import { Ingredient } from "./ingredient";
import { Section } from "./section";
import { Photo } from "./photo";

export interface Recipe {
  id: number;
  title: string;
  description: string;
  createdAt: Date;
  ingredients: Ingredient[];
  sections: Section[];
  photos: Photo[];
}

File: webapi/Entities/Recipe.cs
#nullable disable

using System.ComponentModel.DataAnnotations;

namespace webapi.Entities
{
	public class Recipe
	{
        public int Id { get; set; }

        [Required]
        public string Title { get; set; }

        public string Introduction { get; set; }

        public DateTime CreatedAt { get; set; }

        public List<Ingredient> Ingredients { get; set; }

        public List<Section> Sections { get; set; }

        public List<Photo> Photos { get; set; }
    }
}
